src/vivarium_public_health/disease/state.py

```diff
@@ -465,19 +462,22 @@ def setup(self, builder: Builder) -> None:
         self.disability_weight = self.get_disability_weight_pipeline(builder)
 
         builder.value.register_value_modifier(
-            "all_causes.disability_weight", modifier=self.disability_weight
+            "all_causes.disability_weight",
+            modifier=self.disability_weight,
+            component=self,
         )
 
         self.has_excess_mortality = is_non_zero(
             self.lookup_tables["excess_mortality_rate"].data
         )
-        self.excess_mortality_rate = self.get_excess_mortality_rate_pipeline(builder)
         self.joint_paf = self.get_joint_paf(builder)
+        self.excess_mortality_rate = self.get_excess_mortality_rate_pipeline(builder)
 
         builder.value.register_value_modifier(
             "mortality_rate",
             modifier=self.adjust_mortality_rate,
-            requires_values=[self.excess_mortality_rate_pipeline_name],
+            component=self,
+            required_resources=[self.excess_mortality_rate],
         )
 
         self.randomness_prevalence = self.get_randomness_prevalence(builder)
```

Comment:
This was previously not being done, but technically, `DiseaseState` should add `self.randomness_prevalence` to its `initialization_requirements`. So you should override that property here like:
```python
def initialization_requirements(self) -> list[str | Resource]:
    return super().initialization_requirements() + [self.randomness_prevalence]
```